### ISSUE TITLE
Error when using Godep to manage dependencies

### ISSUE THREAD

**Issue by jiatongw** [kubernetes/kubernetes#66650]
I am working on some code that need a dependency, which is already one of the vendors in kubernetes. **I fetch the latest version and manually copy it** to `kubernetes/vendor` to replace the older version.  (for example, `kubernetes/vendor/github.com/foo` is the older version dep, what I did was I manually fetched the latest version to my GOPATH, and then copy&paste it to `kubernetes/vendor/github.com/foo` to replace the older one)

Then, I typed `hack/run-in-gopath.sh hack/godep-save.sh` to save it. But, error happened:
```
!!! Error in hack/godep-save.sh:72
  Error in hack/godep-save.sh:72. 'GOPATH="${GOPATH}:$(pwd)/staging" godep save "${REQUIRED_BINS[@]}"' exited with status 1
Call stack:
  1: hack/godep-save.sh:72 main(...)
Exiting with status 1
!!! [0726 00:21:23] _tmp/godep-save.15578/vendor exists, restoring it
!!! [0726 00:21:23] _tmp/godep-save.15578/Godeps exists, restoring it
!!! Error in hack/run-in-gopath.sh:33
  Error in hack/run-in-gopath.sh:33. '"${@}"' exited with status 1
Call stack:
  1: hack/run-in-gopath.sh:33 main(...)
Exiting with status 1
```
Please correct me if I missed some steps or did something wrong?

**Comment by dims** [comment]
@jiatongw did you see https://github.com/kubernetes/community/blob/master/contributors/devel/godep.md ?

**Comment by jiatongw** [comment]
Hi @dims ! Yes, I did. Cause the dependency is already there and I want to update it. So I assume that I will only do `Updating a dep` and `Saving deps` right? Now I have the latest version in my GOPATH, and I also copied it into vendor folder. But running the commands listed in`Updating a dep` section also failed. (this one`hack/run-in-gopath.sh bash -c 'git -C $GOPATH/src/example.com/go/frob checkout v2.0.0'`. ) 
```
!!! Error in hack/run-in-gopath.sh:33
  Error in hack/run-in-gopath.sh:33. '"${@}"' exited with status 1
Call stack:
  1: hack/run-in-gopath.sh:33 main(...)
Exiting with status 1
```
